docs/source/workflows/function-groups.md

```diff
@@ -17,190 +17,780 @@ limitations under the License.
+If we have a collection of math functions, we can create a function group to share the configuration and context for all the functions in the group.
 
-### Exposing Functions
+**Python code**:
+```python
+class MathGroupConfig(FunctionGroupBaseConfig, name="math_group"):
+    pass
 
-Use the `include` list to control which functions are added to the global registry. If `include` is empty, the group remains usable as a group, but individual functions are not globally addressable. If `exclude` is provided, matching functions are filtered out and are not wrapped as tools, but they remain accessible programmatically.
+@register_function_group(config_type=MathGroupConfig)
+async def build_math_group(config: MathGroupConfig, _builder: Builder):
+    group = FunctionGroup(config=config)
 
-### Dynamic Filtering
+    async def add(a: float, b: float) -> float:
+        return a + b
+    
+    async def multiply(a: float, b: float) -> float:
+        return a * b
 
-Function groups support dynamic filtering to control which functions are accessible at runtime. Filters work alongside the `include` and `exclude` configuration and are applied when functions are accessed, not when they are added.
+    async def divide(a: float, b: float) -> float:
+        if b == 0:
+            raise ValueError("Cannot divide by zero")
+        return a / b
 
-#### Filter Types
+    group.add_function(name="add", fn=add, description="Add two numbers")
+    group.add_function(name="multiply", fn=multiply, description="Multiply two numbers")
+    group.add_function(name="divide", fn=divide, description="Divide two numbers")
 
-**Group-level filters**: Applied to all functions in the group and receive a list of function names to filter.
+    yield group
+```
 
-**Per-function filters**: Applied to individual functions and receive the function name to determine inclusion.
+**Configuration file**:
 
-#### Filter Interaction
+```yaml
+function_groups:
+  math:
+    _type: math_group
+```
 
-Filters work in combination with `include` and `exclude` configuration:
+### Function Naming and Namespacing
 
-1. First, the configuration determines the base set of functions (include/exclude logic)
-2. Then, group-level filters are applied to the resulting set
-3. Finally, per-function filters are applied to each remaining function
+Functions inside a group are automatically namespaced by the group instance name. This creates a clear hierarchy and prevents naming conflicts.
 
-### Tool Wrapping
+**Pattern**: `instance_name.function_name`
 
-You can request tools for an entire function group. The builder will wrap all accessible functions (honoring `include`, `exclude`, and any active filters) into the requested tool wrapper for a given agent framework.
+**Example**: If your group instance name is `math` and you add functions named `add` and `multiply`:
+- Functions become: `math.add` and `math.multiply`
+- These names are used in workflow configurations and when calling functions
```

Comment:
Not clear from this section when to use the function group name itself. i.e. how do you use it when referencing multiple tools